byteslice.go

```diff
@@ -20,6 +21,10 @@ func ReadByteSlice(r io.Reader, lmt int, n *int, err *error) []byte {
 		*err = ErrBinaryReadInvalidLength
 		return nil
 	}
+	if length > math.MaxInt32 {
```

Comment:
a comment would be nice. like `check that length is less than the maximum slice size`

Reply:
We could have also compared to (lmt - *n) here, I think!